src/coreclr/src/vm/diagnosticsprotocol.h

```diff
@@ -103,6 +103,56 @@ namespace DiagnosticsIpc
 
     const MagicVersion DotnetIpcMagic_V1 = { "DOTNET_IPC_V1" };
 
+    /**
+     * ==ADVERTISE PROTOCOL==
+     * Before standard IPC Protocol communication can occur on a client-mode connection
+     * the runtime must advertise itself over the connection.  ALL SUBSEQUENT COMMUNICATION 
+     * IS STANDARD DIAGNOSTICS IPC PROTOCOL COMMUNICATION.
+     * 
+     * See spec in: dotnet/diagnostics@documentation/design-docs/ipc-spec.md
+     * 
+     * The flow for Advertise is a one-way burst of 24 bytes consisting of
+     * 8 bytes  - "ADVR_V1\0" (ASCII chars + null byte)
+     * 16 bytes - random 128 bit number cookie (little-endian)
+     * 8 bytes  - PID (little-endian)
```

Comment:
Lets add a 2 byte size field at the end of this message which currently is set to 0. In the future if we need to add more data to this advertisement (perhaps something security related or more identifying information?) we can set the size field equal to the amount of appended data. An agent receiving this message should skip ahead this many bytes of data. Future versions of the agent that understand the optional data can interpret it.